packages/caver-transaction/src/transactionTypes/abstractTransaction.js

```diff
@@ -0,0 +1,336 @@
+            throw new Error(`The from address of the transaction is different with the address of the keyring to use.`)
+
+        await this.fillTransaction()
+        const hash = hasher(this)
+        const role = this.type.includes('ACCOUNT_UPDATE') ? KEY_ROLE.ROLE_ACCOUNT_UPDATE_KEY : KEY_ROLE.ROLE_TRANSACTION_KEY
+
+        const sig = keyring.signWithKey(hash, this.chainId, role, index)
+
+        this.appendSignatures(sig)
+
+        return this
+    }
+
+    /**
+     * Signs to transaction with multiple private keys.
+     *
+     * @async
+     * @param {Keyring|string} key - The instance of Keyring, private key string or KlaytnWalletKey string.
+     * @param {function} [hasher] - The function to get hash of transaction.
+     * @return {Transaction}
+     */
+    async signWithKeys(key, hasher = TransactionHasher.getHashForSigning) {
+        let keyring = key
+        if (_.isString(key)) keyring = Keyring.createFromPrivateKey(key)
+        if (!(keyring instanceof Keyring))
+            throw new Error(
+                `Unsupported key type. The key must be a single private key string, KlaytnWalletKey string, or Keyring instance.`
+            )
+
+        // When user attempt to sign with a updated keyring into a TxTypeLegacyTransaction error should be thrown.
+        if (this.type === TX_TYPE_STRING.TxTypeLegacyTransaction && keyring.isDecoupled())
+            throw new Error(`A legacy transaction must be signed with a legacy account key.`)
+
+        if (!this.from || this.from === '0x') this.from = keyring.address
+        if (this.from.toLowerCase() !== keyring.address.toLowerCase())
+            throw new Error(`The from address of the transaction is different with the address of the keyring to use.`)
+
+        await this.fillTransaction()
+        const hash = hasher(this)
+        const role = this.type.includes('ACCOUNT_UPDATE') ? KEY_ROLE.ROLE_ACCOUNT_UPDATE_KEY : KEY_ROLE.ROLE_TRANSACTION_KEY
+
+        const sigs = keyring.signWithKeys(hash, this.chainId, role)
+
+        this.appendSignatures(sigs)
+
+        return this
+    }
+
+    /**
+     * Appends signatures to transaction.
+     *
+     * @param {Array.<string>|Array.<Array.<string>>} sig - An array of signatures to append to the transaction.
+     *                                                      One signature can be defined in the form of a one-dimensional array or two-dimensional array,
+     *                                                      and more than one signatures should be defined in the form of a two-dimensional array.
+     */
+    appendSignatures(sig) {
+        if (!_.isArray(sig[0])) sig = [sig]
+        this.signatures = this.signatures.concat(sig)
+    }
+
+    /**
+     * Combines signatures to transaction from RLP-encoded transaction strings.
+     * When combining the signatures of RLP-encoded transaction into a transaction instance,
+     * an error is thrown if the deocded transaction contains different value.
```

Comment:
```suggestion
     * an error is thrown if the decoded transaction contains different value.
```

Reply:
What do you mean by "different value" here?

Reply:
umm.... "different value" means that the informations included in decoded transaction is not matched with informations in transaction instance.